spec/chatgpt.spec.ts

```diff
@@ -100,52 +139,323 @@ describe("ChatGPT Flow", () => {
-    });
-
-    // Assertions for Function Call
-    expect(logEntry?.functionCalls).toEqual([
-      {
-        name: "TEST_WEATHER_FUNC_IMPL",
-        args: undefined,
-        exitCode: undefined,
-        startTime: undefined,
-        endTime: undefined,
-        result: undefined,
-        // TODO: Process these fields.
-      },
-    ]);
+      startTime: StartTime,
+      endTime: EndTime,
+      functionCall: expect.any(Object),
+    });
+
+    // Assertions for Function Call Result
+    expect(logEntry?.functionCallResult).toEqual({
+      startTime: functionCallStartTime,
+      endTime: functionCallEndTime,
+      result: expect.anything(),
+    });
 
     // Assertions for Meta
-    expect(logEntry?.meta).toMatchObject({
-      env: "test",
-      shell: "/bin/zsh",
+    expect(logEntry?.meta).toEqual({
+      totalTokenCount: expect.any(Number),
+      inputTokenCost1k: expect.any(Number),
+      outputTokenCost1k: expect.any(Number),
+      triggerSource: "",
+      userId: "",
+      locale: Intl.DateTimeFormat().resolvedOptions().locale,
+      userTimeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
       operatingSystem: `${os.platform()}/${os.release()}`,
-      // TODO: Add more fields and switch toMatchObject to toEqual.
+      shell: os.userInfo().shell || "Unknown",
+      memory: 0,
+      machineId: expect.any(String),
+      env: "test",
     });
   }, 20000);
 
   it("should process request with failure", async () => {
-    // E.g. by setting TestTopP to value > 1.
+    const TestPrompt = "Deliberate failure test";
+
+    flow.logPrompt(TestPrompt, "user-input");
+
+    const request = createFailureRequest(TestPrompt, []);
+    flow.logRequest(request);
+
+    try {
+      await openai.chat.completions.create(request);
+    } catch (error) {
+      flow.logError(error);
+    }
+
+    const logEntry = await flow.flushLogs();
+
+    // Assertions for Request
+    expect(logEntry?.request).toEqual({
+      prompt: TestPrompt,
+      model: "gpt-4o-mini",
+      temperature: TestTemperature,
+      topP: FailureTestTopP,
+      maxTokens: TestMaxTokens,
+      errorReason: "",
+      functionCalls: undefined,
+      outputMode: undefined,
+      systemPrompt: undefined,
+      tokenCount: undefined,
+      topK: undefined,
+    });
+
+    // Assertions for Meta
+    expect(logEntry?.meta).toEqual({
+      totalTokenCount: undefined,
+      inputTokenCost1k: expect.any(Number),
+      outputTokenCost1k: expect.any(Number),
+      triggerSource: "",
+      userId: "",
+      locale: Intl.DateTimeFormat().resolvedOptions().locale,
+      userTimeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
+      operatingSystem: `${os.platform()}/${os.release()}`,
+      shell: os.userInfo().shell || "Unknown",
+      memory: 0,
+      machineId: expect.any(String),
+      env: "test",
+    });
+
+    // Assertions for Error
+    expect(logEntry?.error).toEqual({
+      error: expect.any(String),
```

Comment:
Can you specify the error here?

Reserve use of `any` for fields that are truly dynamic